server/controllers/tasks/lib/get_new_tasks.js

```diff
@@ -1,33 +1,90 @@
+  const [ newSuggestionsSearchResults, suspectWorksData ] = await Promise.all([
+    searchEntityDuplicatesSuggestions(entity),
     getAuthorWorksData(entity._id),
   ])
-  if (newSuggestions.length <= 0) return []
+  if (newSuggestionsSearchResults.length <= 0) return []
   const { labels: worksLabels } = suspectWorksData
-  return Promise.all(newSuggestions.map(addOccurrencesToSuggestion(suspectWorksData)))
+  const suggestions = await getSuggestionsEntities(newSuggestionsSearchResults)
+
+  const suggestionWithIsbnInWpArticle = await findAuthorsWithIsbnsInWikipediaArticles(suspectWorksData, suggestions)
+  if (suggestionWithIsbnInWpArticle) {
+    return automerge(entity.uri, suggestionWithIsbnInWpArticle.uri)
+  }
+
+  return Promise.all(suggestions.map(addOccurrencesToSuggestion(suspectWorksData)))
   .then(filterOrMergeSuggestions(entity, worksLabels))
   .then(filterNewTasks(existingTasks))
 }
 
-const filterOrMergeSuggestions = (suspect, workLabels) => suggestions => {
+async function getSuggestionsEntities (newSuggestionsSearchResults) {
+  const uris = newSuggestionsSearchResults.map(suggestion => suggestion.uri)
+  const { entities } = await getEntitiesByUris({ uris })
+  return Object.values(entities)
+}
+
+const filterOrMergeSuggestions = (suspect, workLabels) => async suggestions => {
+  const suspectUri = suspect.uri
+  // Merge if entities have a common external identifier
+  const suggestionUriCommonExternalId = await findSuggestionWithSameExternalId(suspect, suggestions)
+  if (suggestionUriCommonExternalId) return automerge(suspectUri, suggestionUriCommonExternalId.uri)
+
   const suspectTerms = getEntityNormalizedTerms(suspect)
   // Do not automerge if author name is in work title
   // as it confuses occurences found on Wikipedia pages
   if (haveExactMatch(suspectTerms, workLabels)) return suggestions
+
   const sourcedSuggestions = filterSourced(suggestions)
   if (sourcedSuggestions.length === 0) return suggestions
   if (sourcedSuggestions.length > 1) return sourcedSuggestions
-  return automerge(suspect.uri, sourcedSuggestions[0])
+  return validateAndAutomerge(suspectUri, sourcedSuggestions[0])
+}
+
+async function findSuggestionWithSameExternalId (suspect, suggestions) {
+  // known case: inv entity had had an externalId before wd item
+  // Using typeSearch results allows to only merge homonyms,
+  // but could be switched to byClaimValue db request (?)
+  const suspectExternalIds = getExternalIdsClaimsValues(suspect.claims)
+  return suggestions.find(suggestion => {
+    const { claims } = suggestion
+    const suggestionExternalIds = getExternalIdsClaimsValues(claims)
+    return isNonEmptyArray(intersection(suggestionExternalIds, suspectExternalIds))
+  })
+}
+
+function getExternalIdsClaimsValues (claims) {
+  const externalIdsClaims = []
+
+  for (const prop in claims) {
+    const values = claims[prop]
+    if (properties[prop]) {
+      const { isExternalId, format } = properties[prop]
```

Comment:
The `isExternalId` flag will be removed when/if https://github.com/inventaire/inventaire/pull/698 is merged. Replacement:
```js
    const { datatype, format } = properties[prop]
    if (datatype === 'external-id') {
```

Reply:
done in eff1dd692